astropy/visualization/units.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+
+__doctest_skip__ = ['quantity_support']
+
+
+import numpy as np
+
+
+def quantity_support():
```

Comment:
I'm not sure I see much benefit to complicating the API there.  I can see using this without `with` as being convenient (and that does work), but if you really want it to consistently turn off, one should use `with` IMHO.